Add landings dates in bug analysis.

La0 opened the pull request:
Hello @marco-c & @calixteman,

As suggested by Ritu, i added the landing dates in bug analysis.
I noticed get_patch_info basically does the same thing: i could use that instead from shipit. What do you think ?

marco-c commented:
There's already some code in `uplift_info` to do this (but it returns "how much time the patch has been on Nightly"), can you add the additional code there?
In theory, we can add "how much time the patch has been on CHANNEL" for channel in CHANNELS (where CHANNELS is all channels or just the channels before the `channel` variable, e.g. if you call `get_uplift_info('beta')` we can consider CHANNELS=['aurora', 'nightly']).

I think this should be enough and the dashboard can show this (how many days has the patch been on a given channel) instead of the landing dates, which might be easier to read.

La0 commented:
I updated the PR to handle landings dates in uplift_info.
I sticked to datetimes (instead of timedelta) because the delta would be against now(), and not against a fixed date (like the next release date): so it would be impractical to test and use in shipit.

marco-c commented on libmozdata/patchanalysis.py:
```diff
@@ -563,6 +563,21 @@ def attachmenthandler(attachments, bugid):
         if info['uplift_comment'] and info['uplift_author']:
             break
 
+    # Landing dates per useful channels
```
Could you move this near the other `landing_delta` code and avoid calling `Bugzilla.get_landing_comments`?

marco-c commented on libmozdata/patchanalysis.py:
```diff
@@ -563,6 +563,21 @@ def attachmenthandler(attachments, bugid):
             break
 
+    # Landing dates per useful channels
+    all_channels = ['nightly', 'aurora', 'beta', 'release']
```
We might need to add `esr` here.

marco-c replied:
I'm not sure, I think in theory it could be either before or after. Perhaps we should just remove the `channels = all_channels[:all_channels.index(channel) + 1]` line and always find the landing dates for all channels.
In ship-it you can then just show the ones that are not None.

La0 replied:
Agreed.

marco-c commented on libmozdata/patchanalysis.py:
```diff
@@ -563,6 +563,21 @@ def attachmenthandler(attachments, bugid):
+        raise Exception('Invalid channel {}'.format(channel))
+    channels = all_channels[:all_channels.index(channel) + 1]
+    landing_patterns = Bugzilla.get_landing_patterns(channels=channels)
+    landing_comments = Bugzilla.get_landing_comments(bug['comments'], [], landing_patterns)
```
No need to call `Bugzilla.get_landing_patterns`. You should be able to simply do `Bugzilla.get_landing_comments(bug['comments'], channels)`.

La0 commented:
I added the support of `esr` in `Bugzilla.get_landing_patterns`, using the latest esr version available.
Also, i simplified the channels selection, and used only one call to `Bugzilla.get_landing_comments`.

marco-c commented on libmozdata/patchanalysis.py:
```diff
@@ -564,13 +564,9 @@ def attachmenthandler(attachments, bugid):
-    landing_patterns = Bugzilla.get_landing_patterns(channels=channels)
-    landing_comments = Bugzilla.get_landing_comments(bug['comments'], [], landing_patterns)
-    landings = dict(zip(channels, [None, ] * 4))
+    channels = ['central', 'nightly', 'aurora', 'beta', 'release', 'esr', ]
```
No need to define both `central` and `nightly`, as they are the same thing.

marco-c commented:
LGTM, just two remaining minor fixes (moving the `landing` code together, so it's easier to see; only using `nightly` in channels, without `central, because they're the same).

marco-c commented:
I clicked the wrong button :/

La0 commented:
Updated the pr
